### PR TITLE
Add support for .ps1 (Powershell) scripts

### PR DESCRIPTION
**Pull request by faerryn**
Note: this requires ‘shell-file-name’, ‘shell-command-switch’, and Powershell’s ExecutionPolicy to be set appropriately.

### CODE REVIEW

**Comment by astoff** [comment]
Thanks for the contribution

> Note: this requires ‘shell-file-name’, ‘shell-command-switch’, and Powershell’s ExecutionPolicy to be set appropriately.

Would it make sense to replace the command with something like `pwsh -c '<what you had before>'`, assuming that it has a `-c` switch like bash and no quoting issues arise? Then it wouldn't matter which shell Emacs uses.

(By the way, do the other `buffer-env-commands` entries work with powershell?)

**Comment by faerryn** [comment]
> Would it make sense to replace the command with something like `pwsh -c '<what you had before>' "$0"`, assuming that it has a `-c` switch like bash and no quoting issues arise? Then it wouldn't matter which shell Emacs uses.

I think that should work, though $0 doesn't have a powershell equivalent (that I know of).
 
> (By the way, do the other `buffer-env-commands` entries work with powershell?)

Powershell can only source ps1 scripts that end with the ps1 extension.  I guess you could read the file and execute the string as code.

**Comment by astoff** [comment]
So do you actually set `shell-file-name` to point to powershell. Now that I think about it, I'd be very surprised that this doesn't break lots of things, since any package that runs shell commands will assume it's a POSIX shell (possible also striving to support the Windows cmd shell).

[By contrast setting `explicit-shell-name` to powershell, which only affects `M-x shell`, should be totally fine.]

In short: I'm interested in adding this feature, but the `buffer-env-commands` entry should be something that works without tweaking `shell-file-name`, that is, assuming that Emacs is using a POSIX shell or the old cmd.exe.

**Comment by astoff** [comment]
Great, thanks!

File: buffer-env.el [hunk 1]
```diff
@@ -68,6 +68,8 @@
     ("guix.scm" . "guix shell -D -f \"$0\" -- env -0")
     ("flake.nix" . "nix develop -c env -0")
     ("shell.nix" . "nix-shell \"$0\" --run \"env -0\"")
+    ("*.ps1" . "powershell -c '& { param($script) . $script > $null; Get-ChildItem env: |\
+                % {\"$($_.Name)=$($_.Value)`0\"} | Write-Host -NoNewLine } '")
```
Comment (astoff) [inline buffer-env.el:72]: Hi @faerryn , can you show me how to change this command so that the script name is read from the second positional parameter? So what before was `powershell -c '...' .envrc` now becomes `powershell -c '...' <ignored> .envrc`?

Comment (faerryn) [reply buffer-env.el:72]: I think if you change the arglist `param($script)` to `param($ignored, $script)` then that should work, but I am really not sure.  Powershell is weird like that.

Comment (astoff) [reply buffer-env.el:72]: Okay, thanks!
